### PR TITLE
CMake: Add tracemerge functionality

### PR DESCRIPTION
**Pull request by dnakamura**
See individual commit messages for details

### CODE REVIEW

**Comment by charliegracie** [comment]
Great job!

**Comment by charliegracie** [comment]
currently no need to run the genie-omr CI since neither of those platforms are using CMake

File: cmake/modules/OmrTracegen.cmake [hunk 1]
```diff
@@ -22,7 +22,15 @@
 if(OMR_TRACEGEN_)
 	return()
 endif()
-set(OMR_TRACEGEN 1)
+
+add_custom_target(run_tracegen)
+
+# Setup a default trace root if one has not alreay been set
+if(NOT DEFINED OMR_TRACE_ROOT)
+	message(WARNING "OMR: No trace root has been set. Defaulting to '${CMAKE_CURRENT_BINARY_DIR}'")
```
[the remark this answers is not in the document]
Comment (dnakamura) [reply cmake/modules/OmrTracegen.cmake]: My logic was that is makes sense to do it in the highest level that included the file.  What if the consumer of OMR (which uses tracegen) is then consumed by a third project (which does care about tracegen)?

[the remark this answers is not in the document]
Comment (rwy0717) [reply cmake/modules/OmrTracegen.cmake]: OK makes sense.

**Pull request merged**